docs/ref/checks.txt

```diff
@@ -507,6 +507,7 @@ The following checks are run if you use the :option:`check --deploy` option:
   should consider enabling this header to protect user privacy.
 * **security.E023**: You have set the :setting:`SECURE_REFERRER_POLICY` setting
   to an invalid value.
+* **security.W023**: :setting:`ALLOWED_HOSTS` must be a list or tuple of strings for deployment.
```

Comment:
It looks that you missed some comments, e.g. https://github.com/django/django/pull/13927#discussion_r588931750

Reply:
My rational was that warnings and errors are separate classes of notification. I'll drop it in though 👍

Reply:
Addressed in https://github.com/django/django/pull/14149/commits/76e427782faaa2a8dbb06af5a8a828577df0c2c3